service/notification/NotificationProcessor.php

```diff
@@ -335,4 +351,32 @@ private function isCurrentOrderStatusANonFinalStatus($currentOrderStatus)
 
         return false;
     }
+
+    /**
+     * @param $notification
+     * @param OrderCore $order
+     * @return bool
+     * @throws \Exception
+     */
+    private function validateWithCartAndOrder($notification, OrderCore $order)
+    {
+        $cart = \Cart::getCartByOrderId($order->id);
+        $cartCurrency = \Currency::getCurrency($cart->id_currency);
```

Comment:
Do the same with the order currency

```suggestion
        $cartCurrency = \Currency::getCurrency($cart->id_currency);
        $orderCurrency = \Currency::getCurrency($order->id_currency);
        $orderCurrencyIso = $orderCurrency['iso_code'];
```